Library/Homebrew/cmd/fetch.rb

```diff
@@ -32,6 +32,7 @@ def fetch
   def fetch_bottle? f
     return true if ARGV.force_bottle? && f.bottle
     return false unless f.bottle && f.pour_bottle?
+    return false if ARGV.build_from_source? || ARGV.build_bottle?
```

Comment:
Moving this above the `if f.file_modified?` saves us having to pass `&& !build_from_source?` here, as Tim suggested.